src/CoreWCF.Primitives/src/CoreWCF/Description/DispatcherBuilder.cs

```diff
@@ -691,9 +691,9 @@ private static bool HaveCommonInitiatingActions(EndpointFilterProvider x, Endpoi
                 serviceHost.Description.Endpoints.Add(serviceEndpoint);
             }
 
-            InitializeServiceHost(serviceHost, services);
             ServiceConfigurationDelegateHolder<TService> configDelegate = services.GetService<ServiceConfigurationDelegateHolder<TService>>();
             configDelegate?.Configure(serviceHost);
+            InitializeServiceHost(serviceHost, services);
```

Comment:
make sense... you will have all behavior context populated and being sent to delegate to modify or add...